### PR TITLE
[docs] Adds Auto, Light, and Dark themes

### PR DESCRIPTION
**Pull request by jonsamp**
# Why

Closes ENG-204

@calebnance opened https://github.com/expo/expo/pull/11573, which added dark mode to the docs. At the same time, I was in the midst of creating a light/dark theme for all of expo's websites, so I implemented a theme system in [@expo/styleguide](https://github.com/expo/styleguide) and applied it to the docs.

# How

I referenced a variety of blog posts about all the edge cases of dark mode:
- https://dev.to/sreetamdas/the-perfect-dark-mode-2d7g
- https://epicreact.dev/css-variables/
- https://www.joshwcomeau.com/react/dark-mode/

I then came up with color palettes and semantic colors for each theme [in figma](https://www.figma.com/file/wbPwIN1kV7wMgMtwHFvZrZ/Design-System?node-id=178%3A38):
Screen Shot 2021-02-22 at 2 54 55 PM

I ended up creating CSS variables that we can swap on the fly (without re-rendering the react tree). I also added a blocking script to run after SSR and before client-side rendering to set the theme, so that there's no flash of white when using dark mode. Finally, I wrapped the app in the theme context to provide all the pages with access to the theme's current name and to listen for system theme updates, so that if your system switches to dark theme in the evening, so will our docs.

After setting up the themes, I replaced all the colors in docs with colors from @expo/styleguide. I believe I got nearly all of them, but since some colors are hard coded/escaped/etc, I can't be 100% sure. I've been clicking through as many pages as I can to find places that need to get fixed. The one thing that isn't theme-able are png images.

# Screenshots

Screen Shot 2021-02-22 at 2 36 57 PM
Screen Shot 2021-02-22 at 2 37 04 PM
Screen Shot 2021-02-22 at 2 37 14 PM

# Test Plan

This affects all the colors in all the docs, so click around and make sure everything looks great in light/dark mode. In addition, change your system's theme with each theme selected. Only on "auto" should the docs change according to your system's theme.

Try out a preview: http://cedrics-expo.s3-website.eu-central-1.amazonaws.com

### CODE REVIEW

File: docs/components/DocumentationFooter.tsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { css } from '@emotion/core';
+import { theme } from '@expo/styleguide';
```
Comment (byCedric) [inline docs/components/DocumentationFooter.tsx:2]: ❤️

File: docs/components/plugins/PlatformsSection.tsx [hunk 2]
```diff
@@ -47,7 +41,7 @@ function getInfo(isSupported: IsSupported, { title }: Platform) {
   } else if (typeof isSupported === 'object') {
     return {
       children: (
-        <a css={STYLES_LINK} target="_blank" href={isSupported.pending}>
+        <a css={STYLES_LINK} target="_blank" href="#">
```
Comment (byCedric) [inline docs/components/plugins/PlatformsSection.tsx:44]: I'm not sure if you intended to change the `href` to `#`, that breaks the "PR Pending" links on [pages like this one](http://cedrics-expo.s3-website.eu-central-1.amazonaws.com/versions/v40.0.0/sdk/amplitude/).

File: docs/components/base/list.tsx [hunk 3]
```diff
@@ -1,10 +1,9 @@
 import { css } from '@emotion/core';
+import { theme } from '@expo/styleguide';
```
Comment (byCedric) [inline docs/components/base/list.tsx:2]: Maybe we can just export the `UL` and `LI` from `@expo/styleguide` for this component (and maybe others)?

**Review by byCedric — APPROVED** [review]
1 small comment about the "PR Pending" links, otherwise LGTM!
giphy

File: docs/components/base/list.tsx (hunk 3 above)
Comment (jonsamp) [reply docs/components/base/list.tsx:2]: I intend to do that for many components, just haven't added them to the styleguide lib yet.

**Pull request merged**